### PR TITLE
OT-CreateGraph | Created a reports graph on the reports page to show user data

### LINKED ISSUE #22: OT-CreateGraph
Create the front-end based off of our mockup on figma: https://www.figma.com/file/kSZKeE1vutPwQwUM6oDvwL/Off-Top?node-id=0%3A1

You will need to create a graph familiar to the one on the figma with static data. Ignore the beats per minute graph for now. Ignore making the navbar that says "hello user" and the sub navbar that says "Sunday -> Monday -> Tuesday"

### PR DESCRIPTION
**Pull request by henrydinhd**
**What I did:** 
-I created a graph on the Reports tab in the nav bar. The graph shows some static data for your heartbeat and offtop score.

**How to test:** 
-Most of you already know this but you will need Zookeeper, Kafka, Spring-boot, and port forwarding running to log into app.
-Once you are able to log into the app, you just need to click on the reports tab and you will see the graph generate. If you click onto another tab and go back to the reports, it will generate again.

### CODE REVIEW

File: lib/reports.dart [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                ),
+              ),
+            ),
+          ),
+      ),
+    );
+  }
+}
+
+class Scores{
```
Comment (frlzjosh) [inline lib/reports.dart]: can you move this class to its own file in components/graph/scores.dart

File: lib/reports.dart [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    _generateData();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: Scaffold(
+      body:
+        SafeArea(
+            child: Container(
```
Comment (frlzjosh) [inline lib/reports.dart]: move everything starting from the Container(...) on line 71 to a new file called something like reportsGraph.dart and move it to components/reports/reportsGraph.dart.

Then you can simply import the file in here and pass the class name that you declare in that file like this:


```
body:
   SafeArea(
      child: ReportsGraph()
   )
```

**Review by frlzjosh — CHANGES_REQUESTED** [review]
Good job, you just need to create components so that we can abstract our code more.

**Review by steve545 — COMMENTED** [review]
It works and the graph looks cool!

File: lib/reports.dart (hunk 2 above)
Comment (frlzjosh) [reply lib/reports.dart]: Here are examples of how we've done this in other dart files. 
- https://github.com/Off-Top-App/off-top-flutter/blob/master/lib/recordingSession.dart
- https://github.com/Off-Top-App/off-top-flutter/blob/master/lib/components/recordingSession/meter.dart

**Review by frlzjosh — CHANGES_REQUESTED** [review]
You have merge conflicts, you need to merge master and fix these before we can approve this PR.

File: lib/components/footer/bottomNavigationTabs.dart [hunk 3]
```diff
@@ -2,6 +2,7 @@ import 'package:flutter/material.dart';
 import 'package:off_top_mobile/login/login.dart';
 import 'package:off_top_mobile/recordingSession.dart';
 import 'package:off_top_mobile/settings_page.dart';
+import '../../reports.dart';
```
Comment (tonynguyen98) [inline lib/components/footer/bottomNavigationTabs.dart]: Can you change this to be `package:off_top_mobile/reports.dart `

File: lib/components/reports/reportsGraph.dart [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            Text(
+              'Focus Score Graph',
+              style: TextStyle(fontSize: 24.0, fontWeight: FontWeight.bold),
+            ),
+            Expanded(
+              child: charts.LineChart(_seriesLineData,
+                  defaultRenderer: new charts.LineRendererConfig(
+                      includeArea: true, stacked: true),
+                  animate: true,
+                  animationDuration: Duration(seconds: 5),
```
Comment (frlzjosh) [inline lib/components/reports/reportsGraph.dart]: it's a bit too slow change it to 1.5 seconds if not then change it to 1 second

File: lib/components/reports/reportsGraph.dart [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      new Scores(5, 70),
+    ];
+
+    var linesscoresdata2 = [
+      new Scores(0, 35),
+      new Scores(1, 46),
+      new Scores(2, 45),
+      new Scores(3, 50),
+      new Scores(4, 51),
+      new Scores(5, 60),
```
Comment (tonynguyen98) [inline lib/components/reports/reportsGraph.dart]: With Flutter 1.17, we don't need to use the `new` keyword anymore.

File: lib/components/reports/reportsGraph.dart [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+      child: Center(
+        child: Column(
+          children: <Widget>[
+            Text(
+              'Focus Score Graph',
+              style: TextStyle(fontSize: 24.0, fontWeight: FontWeight.bold),
+            ),
+            Expanded(
+              child: charts.LineChart(_seriesLineData,
+                  defaultRenderer: new charts.LineRendererConfig(
```
Comment (tonynguyen98) [inline lib/components/reports/reportsGraph.dart:75]: remove new

File: lib/components/reports/reportsGraph.dart [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+                      includeArea: true, stacked: true),
+                  animate: true,
+                  animationDuration: Duration(seconds: 5),
+                  behaviors: [
+                    new charts.SeriesLegend(
```
Comment (tonynguyen98) [inline lib/components/reports/reportsGraph.dart]: remove new

File: lib/components/reports/reportsGraph.dart [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+                      position: charts.BehaviorPosition.bottom,
+                      outsideJustification:
+                          charts.OutsideJustification.middleDrawArea,
+                      horizontalFirst: false,
+                      desiredMaxRows: 1,
+                      cellPadding: new EdgeInsets.only(right: 100.0,bottom: 5.0, top: 10.0),
```
Comment (tonynguyen98) [inline lib/components/reports/reportsGraph.dart]: You can use the replace feature to find `new` and replace with an empty space

**Review by tonynguyen98 — DISMISSED** [review]
Just a couple easy changes. Was able to build the app and get the graphs to update. Looks great!

File: lib/reports.dart [hunk 9]
```diff
@@ -10,8 +14,16 @@ class ReportsPage extends StatefulWidget {
 class _ReportsState extends State<ReportsPage> {
   @override
   Widget build(BuildContext context) {
-    return Container(
-      child: UserReportTable(),
-    );
+    return Scaffold(
+        body: SafeArea(
+          child: SingleChildScrollView(
+      child: Column(
```
Comment (frlzjosh) [inline lib/reports.dart:20]: fix the indentation

**Review by frlzjosh — CHANGES_REQUESTED** [review]
one small change! Please change mine and tony's requests and your code is good to go!

**Review by frlzjosh — APPROVED** [review]
Great work man!

**Pull request merged**